File: solution files/Python/part-02/task-03-example-driven/example_prompting.py
from datetime import datetime
from typing import List


def format_names(names: List[str]) -> List[str]:
    # Example: ["alice", "bob"] -> ["ALICE", "BOB"]
    return [name.upper() for name in names]


def format_dates(dates: List[str]) -> List[str]:
    # Using the same pattern as format_names above:
    # Example: ["2026-01-17", "2026-02-20"] -> ["Jan 17, 2026", "Feb 20, 2026"]
    return [datetime.strptime(d, "%Y-%m-%d").strftime("%b %d, %Y") for d in dates]

File: solution files/Javascript/part-02/task-03-example-driven/examplePrompting.js
// Example: ["alice", "bob"] -> ["ALICE", "BOB"]
const formatNames = (names) => names.map((name) => name.toUpperCase());

// Using the same pattern as formatNames above:
// Example: ["2026-01-17", "2026-02-20"] -> ["Jan 17, 2026", "Feb 20, 2026"]
const formatDates = (dates) => {
    return dates.map((d) =>
        new Date(d).toLocaleDateString("en-US", {
            month: "short",
            day: "numeric",
            year: "numeric",
        })
    );
};

module.exports = { formatNames, formatDates };
